runtime/oti/bcverify.h

```diff
@@ -135,6 +135,14 @@ typedef struct J9BCVAlloc {
 #define CHECK_TEMP_DOUBLE( tempIndex ) \
 	CHECK_TEMP_PAIR( tempIndex , (UDATA) BCV_BASE_TYPE_DOUBLE );
 
+/* Only used for baload and bastore */
+#define CHECK_BOOL_ARRAY(opCode, operandType) \
+	if (opCode == bc) { \
+		inconsistentStack |= ((operandType != arrayType) && (BCV_BASE_ARRAY_TYPE_BOOL != arrayType)); \
```

Comment:
Should this be `||`?

```
A = operandType != arrayType
B = BCV_BASE_ARRAY_TYPE_BOOL != arrayType
 | A | B | && | ||  | 
 | T | T | T  | T   |
 | T | F | F  | T   |
 | F | T | F  | T   |
 | F | F | F  | F   |
```

The stack is consistent only if the operand is an array of type bool.  So its inconsistent if either of those conditions are false.

Reply:
Logically,  it must be && for baload and bastore.

The reason is that the existing implementation prevents us from inserting a new array type (boolean array) to J9JavaBytecodeArrayTypeTable (in which the expected type is determined by J9JavaBytecodeArrayTypeTable[bc - JBiaload or JBiastore]).

e.g. 
const U_32 J9JavaBytecodeArrayTypeTable[] = {
	0x00000022 ,
	0x00000082 ,
	0x00000042 ,
	0x00000102 ,
	0x00000002 ,
	0x00000402 ,  <-------  BCV_BASE_ARRAY_TYPE_BYTE is  the expected type is byte array for baload and bastore
	0x00000802 ,
	0x00000202 ,  };

For  baload and bastore, there is no way to determine whether the target type (popped out from the stack) is byte array (calculated via J9JavaBytecodeArrayTypeTable)  or boolean array (not in J9JavaBytecodeArrayTypeTable).

To be specific, the macro can be expended to:
if (JBiaload or  JBiastore == bc) {
inconsistentStack |= (BCV_BASE_ARRAY_TYPE_BYTE != arrayType) && (BCV_BASE_ARRAY_TYPE_BOOL != arrayType)

which means, if the bytecode is bload/bastore, the type on the top of the stack must be byte array or boolean array; otherwise, a VerifiyError should be thrown out in such case.

Reply:
Can you record this in a comment in the code?  It will be much easier to understand for the next person to modify this code.

Reply:
Already added comments on the top of the macro.